### ISSUE TITLE
batterymonitor@pdcurtis: determining editor is fragile

### ISSUE THREAD

**Issue by spantaleev** [linuxmint/cinnamon-spices-applets#1985]
* Applet version/Build date: 1.3.2
 * Cinnamon version (cinnamon --version): 3.8.9
 * Distribution: Archlinux

**Notify author of applet**
@pdcurtis 

**Issue**

Pressing the "Open stylesheet (Advanced Function)" button doesn't work.

Looking at the code, it seems like it checks the Cinnamon version and decides to use `gedit` (if the version is below 3.0) or `xed` otherwise.

In my case (Cinnamon 3.8.9), it would try to use `xed`, which is not available.
As a workaround, one could do: `ln -s /usr/bin/gedit /usr/local/bin/xed` and fake having `xed`.

What would be better, though, is if the extension didn't rely on version detection, but rather just used `xdg-open`.
I'm not sure if `xdg-open` (`xdg-utils`) is something that the Cinnamon desktop can be sure is available though. It doesn't seem like it's a required dependency for it.

Alternatively, it could check if the `xdg-open` program is available.. And then fall-back to checking other stuff.
It seems like the extension checks for things like `sox` and `zenity` using `GLib.find_program_in_path()`, so why not use that to check for `xdg-open`, `gedit`, `xed`, etc.

**Comment by pdcurtis** [comment]
@spantaleev Yes this is an issue i am aware of. In fact I have had branches ready for PRs for a number of applets ready to change to xdg-open fora while but, as you say, one can not be any more sure it will be available than xed. As far as I can tell every distribution which has implemented Cinnamon has also added the x utilities so I finally decided to leave be on the basis that "If it ain't broken don't fix it". I do plan to add a further note in the help files. The use of xdg-open has another problem that one does not know what program will be used and markdown files can be opened in all sorts of editors which is not the intention for some of the files.
In summary I do not plan a high priority fix to this or other applets but this should receive a Feature Request Tag so I do not forget!
